src/shared/vscode/commands2.ts

```diff
@@ -306,23 +326,31 @@ class CommandResource<T extends Callback = Callback, U extends any[] = any[]> {
 }
 
 interface CommandInfo<T extends Callback> {
-    readonly label: string
-    readonly args: Parameters<T>
+    readonly id: string
+    readonly name?: string
+    /** Display label, generated from id + name. */
+    readonly label?: string
+    readonly args?: Parameters<T>
     readonly logging?: boolean
+    /** Does the command require credentials? (default: true) */
+    readonly autoconnect?: boolean
     readonly errorHandler?: (error: Error) => ReturnType<T> | void
 }
 
 async function runCommand<T extends Callback>(fn: T, info: CommandInfo<T>): Promise<ReturnType<T> | undefined> {
     const { args, label, logging } = { logging: true, ...info }
     const logger = logging ? getLogger() : new NullLogger()
-    const withArgs = args.length > 0 ? ` with arguments [${args.map(String).join(', ')}]` : ''
+    const withArgs = args && args.length > 0 ? ` with arguments [${args.map(String).join(', ')}]` : ''
 
     // TODO(sijaden): add telemetry instrumentation here
 
     logger.debug(`command: running ${label}${withArgs}`)
 
     try {
-        const result = await fn(...args)
+        if (info.autoconnect === undefined || info.autoconnect === true) {
+            await vscode.commands.executeCommand('aws.tryAutoConnect')
```

Comment:
this doesn't block :(

Reply:
I gave up on this and just moved the code to LoginManager.ts (which is better anyway :)